### PR TITLE
func: init at 3.0.3568

### LINKED ISSUE #118708: azure-function-core-tools
**Project description**
Command line tools for Azure Functions

Existing package from @argylelabcoat here:

https://github.com/argylelabcoat/mypkgs/blob/8ceaaa3448e53c9985549c80956cbef5028e013e/tools/admin/azure-functions-core-tools/default.nix

**Metadata**

* homepage URL: https://github.com/Azure/azure-functions-core-tools
* source URL: https://github.com/Azure/azure-functions-core-tools
* license: mit
* platforms: unix

### PR DESCRIPTION
**Pull request by jshcmpbll**
closes #118708 

###### Motivation for this change

Adding azure-functions-core-tools func cli tool.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jshcmpbll** [comment]
closes #118708

**Comment by jshcmpbll** [comment]
I ran `nixpkgs-fmt` on `pkgs/top-level/all-packages.nix`

I'm okay with dropping this and leaving the formatting as is but I think we should make the format consistent and I think this is the best method..

**Review by sarcasticadmin — APPROVED** [review]
@jshcmpbll `func` additional looks good to me 🙂 

> I ran nixpkgs-fmt on pkgs/top-level/all-packages.nix
>
> I'm okay with dropping this and leaving the formatting as is but I think we should make the format consistent and I think this is the best method..

Dont have strong feelings about this either way but checkout: https://github.com/NixOS/nixpkgs/issues/120832 This might be a bit disruptive for PRs still in progress since its touching so many lines (something else to consider). At the very least, I would leave the reformat for `all-packages.nix` as a separate commit outside of the `func: init` commit.

File: pkgs/development/tools/azure-functions-core-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    wrapProgram "$out/bin/func" --prefix LD_LIBRARY_PATH : ${libPath}
+  '';
+  dontStrip = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/Azure/azure-functions-core-tools";
+    description = "Command line tools for Azure Functions";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jshcmpbll ];
+    platforms = [ "x86_64-linux" "i686-linux" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/azure-functions-core-tools/default.nix:73]: Why not just linux?

File: pkgs/development/tools/azure-functions-core-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r /build/src $out/bin
+    patchelf \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${libPath}" "$out/bin/func"
+    chmod +x $out/bin/func
+    chmod +x $out/bin/gozip
```
Comment (SuperSandro2000) [inline pkgs/development/tools/azure-functions-core-tools/default.nix:62]:
```suggestion
    chmod +x $out/bin/func $out/bin/gozip
```

File: pkgs/development/tools/azure-functions-core-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    find $out/bin -type f -name "*.so" -exec patchelf --set-rpath "${libPath}" {} \;
+    wrapProgram "$out/bin/func" --prefix LD_LIBRARY_PATH : ${libPath}
+  '';
+  dontStrip = true;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/azure-functions-core-tools/default.nix:66]: Why not?

File: pkgs/development/tools/azure-functions-core-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    libuuid
+    stdenv.cc.cc
+    curl
+    zlib
+    icu
+    openssl
+  ];
+
+  unpackPhase = ''
+    unzip -d /build/src $src
```
Comment (SuperSandro2000) [inline pkgs/development/tools/azure-functions-core-tools/default.nix:52]: Please do not hardcode /build

**Review by SuperSandro2000 — COMMENTED** [review]
Please revert all formatting changes in all-packages.

**Pull request merged**